common/persistence/elasticsearch/esProcessor.go

```diff
@@ -211,12 +220,21 @@ func (p *esProcessorImpl) bulkAfterAction(_ int64, requests []elastic.BulkableRe
 		if visibilityTaskKey == "" {
 			continue
 		}
+		if i >= len(response.Items) {
```

Comment:
is it always guaranteed that the last response is missing? What if the first response is missing? If that is the case, isn't it possible that we accidentally correlate the 2nd response with the first request?

Is there a way to directly correlate a response item with the exact request object?